src/Datadog.Trace/RuntimeMetrics/AzurePerformanceCountersListener.cs

```diff
@@ -0,0 +1,81 @@
+// <copyright file="AzurePerformanceCountersListener.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using Datadog.Trace.Util;
+using Datadog.Trace.Vendors.Newtonsoft.Json;
+using Datadog.Trace.Vendors.StatsdClient;
+
+namespace Datadog.Trace.RuntimeMetrics
+{
+    internal class AzurePerformanceCountersListener : IRuntimeMetricsListener
```

Comment:
PerformanceCounters is .NET Framework only right? Should this be .NET Framework only too? I know we _currently_ don't support Linux in AAS anyway, but may be better to be explicit anyway.

Also, is there value in also extracting other aspnet releted metrics ([as we do in `RuntimeEventListener`](https://github.com/DataDog/dd-trace-dotnet/blob/master/src/Datadog.Trace/RuntimeMetrics/RuntimeEventListener.cs#L49-L58)) seeing as they're exposed here (i.e. use `WEBSITE_COUNTERS_ALL`?

Reply:
> Should this be .NET Framework only too?
Good point. Done in https://github.com/DataDog/dd-trace-dotnet/pull/1651/commits/dc217219603dc15b65cf4417e713c1a3de64e81e

> Also, is there value in also extracting other aspnet releted metrics
Currently we rely on the agent to collect aspnet metrics. Maybe it would be worth changing that, but it's beyond the scope of this PR.

Reply:
May want to name this `AzureAppServicePerformanceCounters` for less ambiguity